vault/mount.go

```diff
@@ -960,7 +962,7 @@ func (c *Core) handleDeprecatedMountEntry(ctx context.Context, entry *MountEntry
 
 		case consts.PendingRemoval:
 			dl.Error(errDeprecatedMount.Error())
-			if allow := os.Getenv(consts.VaultAllowPendingRemovalMountsEnv); allow == "" {
+			if !PendingRemovalMountsAllowed {
```

Comment:
should we not get PendingRemovalMountsAllowed from consts.VaultAllowPendingRemovalMountsEnv?

Reply:
The logic to initialize this var is here: https://github.com/hashicorp/vault/pull/17319/files#diff-de23c936a9ff46fc796a96721e088734a0d6281da4f7c0fe19c3ca6fc86c5dc4R1270. This uses the const, which is `VAULT_ALLOW_PENDING_REMOVAL_MOUNTS`.

Reply:
We could also do away with the var if you prefer, but I figured this would be cleaner/easier to track down, since the var is now processed with the associated `server` command.